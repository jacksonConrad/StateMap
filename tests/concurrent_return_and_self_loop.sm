void DFA v_return() {
	start {
		return;
	}
}

void DFA i_return() {
	start {
		return 1;
	}	
}

void DFA str_return() {
	start {
		return "1";
	}	
}

void DFA main()	
{
	stack<string> s;
	start	
	{
		s = concurrent(v_return(), i_return(), str_return());
		pop_con <- *;
	}

	pop_con {
		print(s.pop());
		pop_con <- EOS != s.peek();
		finish <- *;
	}

	finish {
		print("Finish");
		return;
	}
}
