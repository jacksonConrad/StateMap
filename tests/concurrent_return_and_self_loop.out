1
1
1
Finish
